envi/archs/arm/disasm.py

```diff
@@ -218,7 +218,7 @@ def p_dp_imm_shift(opval, va):
 qop_mnem = ('qadd','qsub','qdadd','qdsub')
 smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
 smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
-smul_mnem = ('smulbb','smulbt','smultb','smultt',)
+smul_mnem = ('smulbb','smultb','smulbt','smultt',)  #bt and tb are backwards. Note xy bits are actually yx
```

Comment:
Ok will do.

Date: Fri, 3 Jun 2016 06:55:56 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/archs/arm/disasm.py:

> @@ -218,7 +218,7 @@ def p_dp_imm_shift(opval, va):
>  qop_mnem = ('qadd','qsub','qdadd','qdsub')
>  smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
>  smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
> -smul_mnem = ('smulbb','smulbt','smultb','smultt',)
> +smul_mnem = ('smulbb','smultb','smulbt','smultt',)  #bt and tb are backwards. Note xy bits are actually yx

here is correct.  i'm telling you to swap the letters in the name of the variable in the decoder.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.